### PR TITLE
Allow exporting config and debug.txt with one click (on all platforms) 

### PR DESCRIPTION
**Pull request by grorp**
For fixing bugs reported by users, you often need their `minetest.conf` and `debug.txt`. However, it can be difficult for users to obtain these files. Desktop users have to find out the location of the Minetest user data directory on their platform. Android users can export `debug.txt` with a button in the app, but getting `minetest.conf` requires a computer with ADB installed if you have a recent Android version.

This PR adds a button for exporting so-called "debug info" to the about tab. This button exists on all platforms. On desktop, it copies to the clipboard, on Android, it opens a share pop-up.

This is how the new about tab looks on desktop:
new about tab on desktop

This is how the new about tab looks on Android:
new about tab on android

The so-called "debug info" consists of Minetest version, platform, active renderer, `minetest.conf` and `debug.txt`. Here's an example:

```
Minetest 5.8.0-dev-debug-1890e2c3-dirty
Platform: Android
Active renderer: OpenGL ES 3.2 V@0530.0 (GIT@8382da08aa, I139f36ad46, 1645197494) (Date:02/18/22)

minetest.conf
-------------
screen_w = 2412
font_size = 12
enable_3d_clouds = true
gui_scaling = 1.125
screen_h = 1080
maintab_LAST = about
menu_last_game = minetest

debug.txt
---------


-------------
  Separator
-------------

2023-06-18 19:36:57: ERROR[Main]: Game specified in default_game [minetest] is invalid.


-------------
  Separator
-------------

2023-06-18 19:45:54: ERROR[Main]: Game specified in default_game [minetest] is invalid.
```

Inspired by the discussion starting at https://irc.minetest.net/minetest-dev/2023-06-14#i_6091167. Builds on top of https://github.com/minetest/minetest/pull/12370.

## To do

This PR is a Ready for Review.

## How to test

Look at the about tab, verify that it looks as shown in the screenshots. Press the "Copy debug info" / "Share debug info" button and verify that the output is useful for bug reports.

## Q&A

-   Where did the active renderer text go?

    I had to remove it to make space for the new button. However, I think that the active renderer isn't relevant for the average user anyway. If you really need this information, you can still get it via the "Copy debug info" button.

-   Why does the "minetest.net" button now use the full available width?

    Because the text on this button was cut off on my Android device.

### CODE REVIEW

**Comment by nerzhul** [comment]
it's a very cool idea but i prefer calling it "Export debug info"

**Comment by rubenwardy** [comment]
It should be called "Share" or "Send" debug info on Android, as those are the specific terms on Android.

On desktop, "Copy" makes the most sense as that's what it's done - see above reply

**Comment by grorp** [comment]
> It should be called "Share" or "Send" debug info on Android, as those are the specific terms on Android.

It's currently called "Share debug info" on Android, exactly as you're suggesting.

**Comment by rubenwardy** [comment]
Please rebase, this is near the top of my review list

File: builtin/mainmenu/settings/dlg_settings.lua [hunk 1]
```diff
@@ -378,10 +378,8 @@ local function get_formspec(dialogdata)
 		height = TOUCHSCREEN_GUI and (10 - extra_h) or 12,
 	}
 
-	local scrollbar_w = TOUCHSCREEN_GUI and 0.6 or 0.4
-
 	local left_pane_width = TOUCHSCREEN_GUI and 4.5 or 4.25
-	local search_width = left_pane_width + scrollbar_w - (0.75 * 2)
+	local search_width = left_pane_width + SCROLLBAR_W - (0.75 * 2)
```
Comment (rubenwardy) [inline builtin/mainmenu/settings/dlg_settings.lua:382]: unrelated changes

**Review by rubenwardy — CHANGES_REQUESTED** [review]
I wonder whether this should limit the lines of debug.txt to 30 lines

File: builtin/mainmenu/tab_about.lua [hunk 2]
```diff
@@ -119,13 +119,44 @@ local function build_hacky_list(items, spacing)
...
+	local irrlicht_device = core.get_active_irrlicht_device()
+	local renderer = core.get_active_renderer()
+
+	local path_prefix = core.get_user_path() .. DIR_DELIM
+	local minetest_conf = read_text_file(path_prefix .. "minetest.conf") or "[not available]\n"
+	local debug_txt = read_text_file(path_prefix .. "debug.txt") or "[not available]\n"
+
+	return table.concat({
+		version.project, " ", (version.hash or version.string), "\n",
+		"Platform: ", PLATFORM, "\n",
```
Comment (rubenwardy) [inline builtin/mainmenu/tab_about.lua:143]: Using

```c++
RenderingEngine::get_raw_device()->getOSOperator()->getOperatingSystemVersion()
```

Gives you:

```
Linux 6.4.12-arch1-1 #1 SMP PREEMPT_DYNAMIC Thu, 24 Aug 2023 00:38:14 +0000 x86_64
```

Distro and X11 vs Wayland would be nice as well

**Comment by rubenwardy** [comment]
We should try to get as many things from the [bug report template](https://github.com/minetest/minetest/issues/new?assignees=&labels=Unconfirmed+bug&projects=&template=bug_report.yaml) into here as possible, which includes:

* Architecture
* Build flags (This is already shown in the pause menu, so maybe see there).
	* This would include Lua vs LuaJit, and other dep info
* CPU and GPU info

That last one will probably require porting code, so feel free to skip

File: builtin/mainmenu/settings/dlg_settings.lua (hunk 1 above)
Comment (rubenwardy) [reply builtin/mainmenu/settings/dlg_settings.lua:382]: Nevermind, this is done in the about tab too

File: builtin/mainmenu/tab_about.lua (hunk 2 above)
Comment (grorp) [reply builtin/mainmenu/tab_about.lua]: `IOSOperator::getOperatingSystemVersion` doesn't seem to be that great. Not sure if it returns anything useful on non-Linux platforms. With the SDL device, it only returns the SDL version.

**Comment by grorp** [comment]
> Architecture

> Build flags

Done.

> CPU and GPU info
> That last one will probably require porting code, so feel free to skip

Skipping :)

**Comment by grorp** [comment]
`string.tail` is now a public API. If that's not desired, what should I do instead? Leave it undocumented? Move it to a different file?

I think the "debug info" now contains enough information to be useful. Would it be possible to merge this as-is and, if someone wants to do that, add more information in future PRs?

**Comment by Zughy** [comment]
@grorp rebase needed

File: builtin/common/misc_helpers.lua [hunk 3]
```diff
@@ -195,6 +195,27 @@ function string.split(str, delim, include_empty, max_splits, sep_is_pattern)
 	return items
 end
 
+--------------------------------------------------------------------------------
+function string.tail(str, num_lines)
```
Comment (appgurueu) [inline builtin/common/misc_helpers.lua:199]: Not very important, but if we're gonna expose this, I'd probably try to give it a clearer name ("tail" may also be in reference to characters rather than lines). I'd also probably implement this using a reverse numeric for: `for index = #str, 1, -1 do local char = str:sub(i, i)`.

**Comment by grorp** [comment]
There's a conceptual problem with this PR: Do we actually want minetest.conf/debug.txt at the beginning of every issue, together with the version information? While this would be useful for many bugs, it would be unnecessary noise for some issues, especially as you'd have to scroll down to reach the actual issue content. Having minetest.conf/debug.txt attached as downloadable files could be better.

If you agree, I can think of three options:

1. Edit the issue template to instruct users to retrieve and upload minetest.conf/debug.txt using the "Open user data directory" button (on Android, include minetest.conf in the output of the existing "Share debug log" button)

2. Automatically upload minetest.conf and debug.txt to some filesharing provider when the "Copy debug info" button is clicked and include links in the copied text (privacy?)

4. Simply remove the minetest.conf/debug.txt-related stuff from this PR and limit this PR to copying version information, which is conceptually simple and especially useful because it includes stuff like the Irrlicht device

**Comment by Zughy** [comment]
Maybe option 3? Option 2 sounds like a headache privacy-wise, not sure about option 1. Regardless, I agree that minetest.conf and debug.txt aren't needed most of the times. Minimal tests are way more important

**Comment by grorp** [comment]
I'll go with option 3 and open a new PR with https://github.com/grorp/minetest/tree/about-tab-4 some day

**Comment by rubenwardy** [comment]
I just want to note that the current share debug log button is the only good way to get the debug log on Android, without needing a computer. So that needs to remain
